server/common/utils/type_conversion_utils.py

```diff
@@ -97,6 +97,9 @@ def _get_type_info_from_dtype(dtype) -> Union[Tuple[np.dtype, dict], None]:
     if dtype.kind == "O" and dtype.name != "category":
         return (np.dtype(str), {"type": "string"})
 
+    if allow_int64 and dtype.kind in ["i", "u"] and np.can_cast(dtype, np.int64):
```

Comment:
This will never execute, because:

```
    if dtype.kind in ["i", "u"] and np.can_cast(dtype, np.int32):
        return (np.int32, {"type": "int32"})
```
will always execute before the allow_int64 block can execute.

Note that any integer dtype can be cast into int32 (including np.int64).